addons/persistence/mongodb-persistence-addon/src/main/java/org/kie/kogito/mongodb/DocumentMarshallingStrategy.java

```diff
@@ -19,76 +19,67 @@
 import java.io.ObjectInputStream;
 import java.io.ObjectOutputStream;
 
-import org.kie.api.marshalling.ObjectMarshallingStrategy;
-import org.kie.kogito.mongodb.utils.CommonUtils;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.JsonMappingException;
+import org.kie.api.marshalling.ObjectMarshallingStrategy;
+import org.kie.kogito.mongodb.utils.CommonUtils;
 
 public class DocumentMarshallingStrategy implements ObjectMarshallingStrategy {
-	
-    private static final Logger LOGGER = LoggerFactory.getLogger(DocumentMarshallingStrategy.class);
-    
-	public String marshalToJson(Object object) {
 
-		String json = null;
-		try {
-			json = CommonUtils.getObjectMapper().writeValueAsString(object);
-		} catch (JsonProcessingException e) {
-			 LOGGER.error("Error while writing object as json", e);
-		}
-		return json;
+    @Override
+    public String marshalToJson(Object object) {
+        try {
+            return CommonUtils.getObjectMapper().writeValueAsString(object);
+        } catch (JsonProcessingException e) {
+            throw new IllegalArgumentException("JsonProcessingException writing object as json : " + e.getMessage(), e);
+        }
```

Comment:
Why catch a checked exception and throw an unchecked one? I think its better to have this kind of methods throwing checked exceptions and letting the callers decide what to do.
Besides, I don't think JsonProcessingException means a bad argument. If you want to hide this type of exception you can create a checked exception called MarshalException.

```suggestion
        } catch (JsonProcessingException e) {
            throw new MarshalException("Unable to write object as json", e);
        }
```